### PR TITLE
add ensure_ascii parameter to dump. #330

### PR DESCRIPTION
**Pull request by ryanjmulder**

### CODE REVIEW

File: src/json.hpp [hunk 1]
```diff
@@ -9017,38 +9039,41 @@ class basic_json
...
-    parameter.
+    and @a ensure_ascii parameters.
 
     @param[in] indent If indent is nonnegative, then array elements and object
     members will be pretty-printed with that indent level. An indent level of
     `0` will only insert newlines. `-1` (the default) selects the most compact
     representation.
     @param[in] indent_char The character to use for indentation if @a indent is
     greater than `0`. The default is ` ` (space).
+    @param[in] ensure_ascii If ensure_ascii is true (the default), all non-ASCII
```
Comment (gregmarr) [inline src/json.hpp]: The default in the function declaration is false.

File: src/json.hpp [hunk 2]
```diff
@@ -9017,38 +9039,41 @@ class basic_json
...
+    @param[in] ensure_ascii If ensure_ascii is true, all non-ASCII characters
+    in the output are escaped with \uXXXX sequences, and the result consists
+    of ASCII characters only.
 
     @return string containing the serialization of the JSON value
 
     @complexity Linear.
 
     @liveexample{The following example shows the effect of different @a indent
-    parameters to the result of the serialization.,dump}
```
Comment (nlohmann) [inline src/json.hpp]: The `,` here is required. It separates the description (`The following example...`) from the name of the example `dump` (see file [`doc/examples/dump.cpp`](https://github.com/nlohmann/json/blob/develop/doc/examples/dump.cpp)).

Please undo this change.

**Comment by nlohmann** [comment]
Thanks for the PR! Let's wait for Travis/Coveralls to finish.

**Comment by nlohmann** [comment]
Thanks a lot!

**Comment by ryanjmulder** [comment]
Happy to help! The library is wonderful and it made my project a joy to
program. Thank you!

**Comment by nlohmann** [comment]
@ryanjmulder Unfortunately, I found an error in the code, see #656. Maybe you have an idea how to fix it.
